src/edu/pugetsound/mathcs/nlp/architecture_nlp/features/FoldCreator.java

```diff
@@ -0,0 +1,211 @@
+	    		String rawText = peices[1].substring(8);
+	    		String[] parse = rawText.split("pos");
+	    		
+	    		
+	    		String[] types = new String[parse.length-1]; //types is an array with the word types inside
+	    		
+	    		int iterator = -1; 
+	    		
+	    		for(String x : parse)
+	    			{
+	    				if (iterator!=-1) 
+	    					{
+	    						types[iterator]=(x.split(",")[0].substring(1)); // get the word types of the given sentence
+	    					}
+	    				iterator +=1;
+	    			}
+	        	
+	        	
+	        	
+				List<String>[] outer = new List[types.length]; // create an array of arrayLists
+				
+				for (int i = 0; i < types.length; i++) {
+				    outer[i] = new ArrayList<>(); // fill it
+				}
+				
+				int iterator2 = 0;
+				
+	            for (String x : types) // fill the array of arrayLists with words of the corresponding type in the corresponding arrays
+	            {
+	            
+	        
+	            	for (String y : newWords)
+	            	{
+	            		
+	            		if(y.split("\\.")[1].equals(x))
+	            		{
+	            			outer[iterator2].add(y.split("\\.")[0]);
+	            		}
+	            	}
+	            	iterator2 +=1;
+	            	
+	            	
+	            }
+	            
+				ArrayList<String> newSentances = new ArrayList<String>(); // prepare for new sentences
+				
+				// if the first word is a verb, keep it and start new sentences with it
+	        	if(types[0].equals("VB") || types[0].equals("VBD") || types[0].equals("VBG") || types[0].equals("VBN") || types[0].equals("VBP") || types[0].equals("VBZ") || types[0].equals("VH") || types[0].equals("VHD") || types[0].equals("VHG") || types[0].equals("VHN") || types[0].equals("VHP") || types[0].equals("VHZ") || types[0].equals("VV") || types[0].equals("VVD") ||types[0].equals("VVG") || types[0].equals("VVN") || types[0].equals("VVP") || types[0].equals("VVZ"))
+	        	{
+	        		newSentances.add(rawText.split("token=")[1].split(",")[0]);
+	        	}
+	        	
+	        	// otherwise begin new possible sentences with corresponding words of the same type
+	        	else {
+	        		
+					for (String thisWord : outer[0])
+					{
+						newSentances.add(thisWord);
+					}
+	        	}
+	
+				Random rand = new Random();
+				
+				for (int x = 1; x < types.length; x++) // for each word place 
+	            {
+	    			ArrayList<String> newList = new ArrayList<String>(); // make a temporary list
+	
+	    			// if it's a verb then keep it
+	            	if(types[x].equals("VB") || types[x].equals("VBD") || types[x].equals("VBG") || types[x].equals("VBN") || types[x].equals("VBP") || types[x].equals("VBZ") || types[x].equals("VH") || types[x].equals("VHD") || types[x].equals("VHG") || types[x].equals("VHN") || types[x].equals("VHP") || types[x].equals("VHZ") || types[x].equals("VV") || types[x].equals("VVD") ||types[x].equals("VVG") || types[x].equals("VVN") || types[x].equals("VVP") || types[x].equals("VVZ"))
```

Comment:
Pull this logic out into a private method that takes an integer and returns true or false. Then call this method above and here.